src/GafferArnold/IECoreArnoldPreview/ShaderAlgo.cpp

```diff
@@ -125,6 +126,55 @@ std::vector<AtNode *> convert( const IECore::ObjectVector *shaderNetwork, const
 					continue;
 				}
 			}
+
+			// Check stringVectorData params to see if they are links to other shaders
+			// There is currently a bit of a problem, in that the one case we want to use this in
+			// is light filters, which are of type NODE, and are connected using AiNodeSetPtr/Array
+			// rather than AiNodeLinkOutput
+			//
+			// Currently just hardcoding to use AiNodeSetArray in the vector case.  This works for
+			// all use cases I'm aware of, but there's no reason you couldn't have a non-array
+			// connection of type NODE or an array connection of type RGB.
+			//
+			// Should we be somehow checking if the parameter type is NODE in both the StringData and
+			// StringVectorData case?  I don't see anything in the API to do this.
+			//
+			// Alternatively, should we use a token other than "link:" to connect in this way?
```

Comment:
You an get the type of a parameter with `AiParamGetType()` as seen [here](https://github.com/ImageEngine/cortex/blob/408bb11ea168b8eef04c8c0886b0859fd8883e78/contrib/IECoreArnold/src/IECoreArnold/ParameterAlgo.cpp#L248). For the array type I think you need to go one further and use `AiParamGetDefault()` and then query the type of the array that is returned. If you want to leave that as a todo that's OK though.

I don't think we should use a separate token - at some point I expect we'll have a nicer ShaderNetwork entity in Cortex, and that will have more explicit links than this stopgap method we're using here. I don't expect those to have more than one type.